rocketpy/Flight.py

```diff
@@ -1800,394 +1819,666 @@ def postProcess(self, interpolation="spline", extrapolation="natural"):
+        x, y = self.x[:, 1], self.y[:, 1]
+        bearing = (2 * np.pi - np.arctan2(-x, y)) * (180 / np.pi)
+        return np.column_stack((self.time, bearing))
+
+    @funcify_method("Time (s)", "Latitude (°)", "linear", "constant")
+    def latitude(self):
+        """Rocket latitude coordinate, in degrees, as a rocketpy.Function of time."""
+        lat1 = np.deg2rad(self.env.lat)  # Launch lat point converted to radians
+
+        # Applies the haversine equation to find final lat/lon coordinates
+        latitude = np.rad2deg(
+            np.arcsin(
+                np.sin(lat1) * np.cos(self.drift[:, 1] / self.env.earthRadius)
+                + np.cos(lat1)
+                * np.sin(self.drift[:, 1] / self.env.earthRadius)
+                * np.cos(np.deg2rad(self.bearing[:, 1]))
+            )
+        )
+        return np.column_stack((self.time, latitude))
+
+    @funcify_method("Time (s)", "Longitude (°)", "linear", "constant")
+    def longitude(self):
+        """Rocket longitude coordinate, in degrees, as a rocketpy.Function of time."""
+        lat1 = np.deg2rad(self.env.lat)  # Launch lat point converted to radians
+        lon1 = np.deg2rad(self.env.lon)  # Launch lon point converted to radians
+
+        # Applies the haversine equation to find final lat/lon coordinates
+        longitude = np.rad2deg(
+            lon1
+            + np.arctan2(
+                np.sin(np.deg2rad(self.bearing[:, 1]))
+                * np.sin(self.drift[:, 1] / self.env.earthRadius)
+                * np.cos(lat1),
+                np.cos(self.drift[:, 1] / self.env.earthRadius)
+                - np.sin(lat1) * np.sin(np.deg2rad(self.latitude[:, 1])),
             )
-            if self.freestreamSpeed[i, 1] < 1e-6:
-                angleOfAttack.append([self.freestreamSpeed[i, 0], 0])
-            else:
-                dotProductNormalized = dotProduct / self.freestreamSpeed[i, 1]
-                dotProductNormalized = (
-                    1 if dotProductNormalized > 1 else dotProductNormalized
-                )
-                dotProductNormalized = (
-                    -1 if dotProductNormalized < -1 else dotProductNormalized
-                )
-                angleOfAttack.append(
-                    [
-                        self.freestreamSpeed[i, 0],
-                        (180 / np.pi) * np.arccos(dotProductNormalized),
-                    ]
-                )
-        self.angleOfAttack = Function(
-            angleOfAttack, "Time (s)", "Angle Of Attack (°)", "linear"
         )
 
-        # Converts x and y positions to lat and lon
-        # We are currently considering the earth as a sphere.
-        bearing = []
-        distance = [
-            (self.x[i][1] ** 2 + self.y[i][1] ** 2) ** 0.5 for i in range(len(self.x))
+        return np.column_stack((self.time, longitude))
+
+    @cached_property
+    def retrieve_acceleration_arrays(self):
```

Comment:
```suggestion
    def retrieve_acceleration_arrays(self):
```

Does this need to be a `@cached_property`?

Reply:
The retrieve_acceleration_arrays property is used 6 times at the Flight.py file.
If we don't use it as a cached_property, what you happen is that the functions inside the `def retrieve_acceleration_arrays()` will be called six times, which can significantly impact performance.

I can imagine that better names to this function can be given, if you present some suggestions we can discuss.

Reply:
Oh okay. This makes sense then, no problems here